app/src/main/java/com/akshay/newsapp/news/ui/adapter/NewsArticlesAdapter.kt

```diff
@@ -16,7 +16,7 @@ import kotlinx.android.synthetic.main.row_news_article.view.*
  * The News adapter to show the news in a list.
  */
 class NewsArticlesAdapter(
-        private val listener: (NewsAdapterEvent) -> Unit
+        private val listener: (NewsArticles) -> Unit
```

Comment:
It's not a good idea to tie the listener to specific type, that's why we had `NewsAdapterEvent` sealed class to make the listener generic.
My recommendation is to do the following:
1. Replace the `NewsArticles` to `NewsAdapterEvent` again
2. Change the `ClickEvent` in `NewsAdapterEvent` to as followed:
```
data class ClickEvent(
     val article: NewsArticles
) : NewsAdapterEvent()
```
3. Now use it as following in the `setOnClickListener`
```
setOnClickListener { listener(NewsAdapterEvent.ClickEvent(newsArticle)) }
```